### PR TITLE
Terraform module to demonstrate Feast setup on GCP

### PR DESCRIPTION
**Pull request by khorshuheng**
Signed-off-by: Khor Shu Heng <khor.heng@gojek.com>

**What this PR does / why we need it**:
This provides a terraform module which demonstrate how to set up Feast on GCP, which the user can refer to in order to write their own terraform module to setup Feast for production use case.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by woop** [comment]
@khorshuheng remember to update the documentation with this installation guide.

**Comment by khorshuheng** [comment]
> @khorshuheng remember to update the documentation with this installation guide.

Yes, i will update the documentation on gitbook once this is merged.

File: infra/terraform/gcp/feast.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      dataproc_cluster_name           = google_dataproc_cluster.feast_dataproc_cluster.name
+      dataproc_project                = var.gcp_project_name
+      dataproc_region                 = var.region
+      dataproc_staging_bucket         = var.dataproc_staging_bucket
+      gcp_service_account_secret_name = var.feast_sa_secret_name
+      gcp_service_account_secret_key  = "credentials.json"
+      feast_core_host                 = "${var.name_prefix}-feast-core"
+      feast_core_port                 = 6565
+      kafka_broker_ip                 = google_compute_address.kafka_broker.address
+    })
```
Comment (oavdeev) [inline infra/terraform/gcp/feast.tf:46]: AWS terrafrom config currently uses HCL+`yamlencode` [to generate the values](https://github.com/feast-dev/feast/blob/master/infra/terraform/aws/helm.tf). I don't feel strongly between that and `templatefile` approach used here, but either way I think it makes sense to use either `yamlencode` or `templatefile` for both for consistency.

**Review by oavdeev — APPROVED** [review]

**Pull request merged**